File: Numbers/binary-to-decimal/binary_to_decimal.js
// Binary to decimal conversion

function binary_to_decimal(binary){
    var decimal = 0;
    for(var i = 0; i < binary.length; i++){
        decimal += binary[i] * Math.pow(2, binary.length - i - 1);
    }
    return decimal;
}

console.log(binary_to_decimal("11111111"));

File: Numbers/binary-to-decimal/binary_to_decimal.cs
using System;

namespace bin_to_dec
{
    class BinaryConversions
    {
        public static double dec = 0;
        public static double binToDec(string bin)
        {
            for (var i = 0; i < bin.Length; i++)
            {
                dec += bin[i] * Math.Pow(2, bin.Length - i - 1);
            }

            return dec;

        }
    }
    class Program
    {
        static void Main(string[] args)
        {
            while (true)
            {
                Console.Write("Binary number: ");
                string binStr = Console.ReadLine();
                Console.WriteLine("{0}", BinaryConversions.binToDec(binStr));
            }
        }
    }
}
